src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -88,6 +85,24 @@ public async Task InvokeAsync(HttpContext context)
 					Headers = responseHeaders
 				};
 
+				if (context.User?.Identity != null && context.User.Identity.IsAuthenticated && context.User.Identity != null
+					&& transaction.Context.User == null)
+				{
+					transaction.Context.User = new User
+					{
+						UserName = context.User.Identity.Name,
+						Id = GetClaimWithFallbackValue(ClaimTypes.NameIdentifier, "sub"),

```

Comment:
I would define a const for `"sub"` - IIUC it's OAuth/OpenID claim type.

Reply:
Yes, those are the OpenID claim types. I moved that to a const type.